sumologic_collectd_metrics/metrics_converter.py

```diff
@@ -144,3 +155,82 @@ def convert_to_metrics(data, data_set, sep):
         metrics.append(metric)
 
     return metrics
+
+
+def process_signalfx_statsd_tags(data):
+    """
+    Process statsd tags embedded in the metric name within the supplied raw data. This function
+    modifies the raw data in-place.
+
+    :param data:
+    :type data: collectd.Values
+    :return:
+    :rtype: None
+    """
+    cleaned_type_instance, new_metadata = parse_statsd_signalfx_metric_name(
+        data.type_instance
+    )
+    data.type_instance = cleaned_type_instance
+    data.meta.update(new_metadata)
+
+
+def parse_statsd_signalfx_metric_name(metric_name):
+    """
+    Parse the metric name and extract metadata from it according to SignalFx' custom extension to
+    the StatsD protocol.
+    See: https://docs.signalfx.com/en/latest/integrations/agent/monitors/collectd-statsd.html#adding-dimensions-to-statsd-metrics
+
+    :param metric_name: The metric name
+    :type metric_name: str
+    :return: A pair consisting of the modified metric name without the metadata, and a dict of
+    extracted metadata.
+    :rtype: (str, dict)
+    """  # pylint: disable=line-too-long
+    matcher = re.compile(r"\[([^\]]+)\]")
+    matches = matcher.findall(metric_name)
+    if len(matches) > 1:
+        # if we find more than one metadata segment, something is probably wrong, and we should
+        # ignore it
+        logger.warning(
+            "Found two metadata segments in metric name `%s`. "
```

Comment:
It is not necessarily `two`, it's `more than one` :)
Can we also say in the warning message what action was taken due to this?

`Found more than one metadata segment in metric %s. Ignoring all segments.` or something like this?